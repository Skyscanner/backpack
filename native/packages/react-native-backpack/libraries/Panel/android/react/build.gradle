apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

android {
  compileSdkVersion rootProject.ext.compileSdkVersion
  buildToolsVersion rootProject.ext.buildToolsVersion


  defaultConfig {
    minSdkVersion rootProject.ext.minSdkVersion
    targetSdkVersion rootProject.ext.targetSdkVersion
    versionCode 1
    versionName "1.0"
  }
}

dependencies{
  implementation "com.facebook.react:react-native:+" // From node_modules
  implementation project(':panel-native')
  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
}
repositories {
  mavenCentral()
}
