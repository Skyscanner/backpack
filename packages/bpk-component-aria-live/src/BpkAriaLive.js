/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import type { ReactElement } from 'react';
// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
import { cssModules } from 'bpk-react-utils';

import STYLES from './BpkAriaLive.module.scss';

const getClassName = cssModules(STYLES);

export const POLITENESS_SETTINGS = {
  off: 'off',
  polite: 'polite',
  assertive: 'assertive',
}; // as const; // Temp disabling TS types due to non TS projects compatibility.

// Temp disabling TS types due to non TS projects compatibility.
// export type PolitenessSetting =
//   typeof POLITENESS_SETTINGS[keyof typeof POLITENESS_SETTINGS];

export type Props = {
  children: ReactElement | string,
  politenessSetting?: $Keys<typeof POLITENESS_SETTINGS>, // PolitenessSetting,
  visible?: boolean,
  className?: string | null,
  [rest: string]: any, // Inexact rest. See decisions/inexact-rest.md
};

const BpkAriaLive = ({
  className = null,
  politenessSetting = POLITENESS_SETTINGS.polite,
  visible = false,
  ...rest
}: Props) => {
  const classNames = getClassName(
    'bpk-aria-live',
    !visible && 'bpk-aria-live--invisible',
    className,
  );

  return <div aria-live={politenessSetting} className={classNames} {...rest} />;
};

BpkAriaLive.defaultProps = {
  className: null,
  politenessSetting: POLITENESS_SETTINGS.polite,
  visible: false,
};

export default BpkAriaLive;
