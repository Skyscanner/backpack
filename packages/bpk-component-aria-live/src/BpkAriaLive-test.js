/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';

import BpkAriaLive, { POLITENESS_SETTINGS } from './BpkAriaLive';
// Temp disabling TS types due to non TS projects compatibility.
// import type { PolitenessSetting } from './BpkAriaLive';

describe('BpkAriaLive', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<BpkAriaLive>Backpack</BpkAriaLive>);
    expect(asFragment()).toMatchSnapshot();
  });

  Object.keys(POLITENESS_SETTINGS).forEach((politenessSetting) => {
    it(`should render correctly with politenessSetting="${politenessSetting}"`, () => {
      const { asFragment } = render(
        // Temp disabling TS types due to non TS projects compatibility.
        // <BpkAriaLive politenessSetting={politenessSetting as PolitenessSetting}>
        <BpkAriaLive politenessSetting={politenessSetting}>
          Backpack
        </BpkAriaLive>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should render correctly with the "visible" prop', () => {
    const { asFragment } = render(<BpkAriaLive visible>Backpack</BpkAriaLive>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should support custom class names', () => {
    const { asFragment } = render(
      <BpkAriaLive className="custom-classname">Backpack</BpkAriaLive>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should support arbitrary props', () => {
    const { asFragment } = render(
      <BpkAriaLive testid="123">Backpack</BpkAriaLive>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
