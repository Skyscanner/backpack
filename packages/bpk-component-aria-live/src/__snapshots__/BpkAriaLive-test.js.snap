// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkAriaLive should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="bpk-aria-live bpk-aria-live--invisible"
  >
    Backpack
  </div>
</DocumentFragment>
`;

exports[`BpkAriaLive should render correctly with politenessSetting="assertive" 1`] = `
<DocumentFragment>
  <div
    aria-live="assertive"
    class="bpk-aria-live bpk-aria-live--invisible"
  >
    Backpack
  </div>
</DocumentFragment>
`;

exports[`BpkAriaLive should render correctly with politenessSetting="off" 1`] = `
<DocumentFragment>
  <div
    aria-live="off"
    class="bpk-aria-live bpk-aria-live--invisible"
  >
    Backpack
  </div>
</DocumentFragment>
`;

exports[`BpkAriaLive should render correctly with politenessSetting="polite" 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="bpk-aria-live bpk-aria-live--invisible"
  >
    Backpack
  </div>
</DocumentFragment>
`;

exports[`BpkAriaLive should render correctly with the "visible" prop 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="bpk-aria-live"
  >
    Backpack
  </div>
</DocumentFragment>
`;

exports[`BpkAriaLive should support arbitrary props 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="bpk-aria-live bpk-aria-live--invisible"
    testid="123"
  >
    Backpack
  </div>
</DocumentFragment>
`;

exports[`BpkAriaLive should support custom class names 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="bpk-aria-live bpk-aria-live--invisible custom-classname"
  >
    Backpack
  </div>
</DocumentFragment>
`;
